torch/_inductor/ir.py

```diff
@@ -8117,6 +8117,17 @@ def create_out_of_place(
             packed.outputs = [packed]
             return packed
 
+    def codegen(self, wrapper):
+        super().codegen(wrapper)
+        # NOTE: It should always be safe to attempt to `del` the output of inplace-collective / wait_tensor right after the op,
```

Comment:
Just to refresh my memory - if we have:
```
buf_0 = ...
buf_1 = torch.ops._c10d_functional.all_reduce_(buf_0)
buf_2 = torch.ops._c10d_functional.wait_tensor(buf_0)?
```

Will the subsequent use of the all-reduce input always be `buf_0`

Reply:
Yes it will - I updated the unit test `test_inductor_all_reduce_single` to show this

Reply:
Is `_CollectiveKernel` always in-place? I am not familiar with collective ops, but I see this class contains both `create_inplace` and `create_out_of_place`.

Reply:
`_CollectiveKernel` can either be in-place or out-of-place. I just added a flag `self.inplace` to make it more clear whether a _CollectiveKernel is in-place.